workload_profiling/src/test/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfoSuite.scala

```diff
@@ -30,17 +29,22 @@ class ApplicationInfoSuite extends FunSuite with Logging {
   val sparkSession = ProfileUtils.createSparkSession
   var apps :ArrayBuffer[ApplicationInfo] = ArrayBuffer[ApplicationInfo]()
   val appArgs = new ProfileArgs(Array("src/test/resources/eventlog_minimal_events"))
-  val fileWriter = new PrintWriter("src/test/resources/workload_profiling")
 
   test("test single event") {
-    var index: Int = 1
-    val eventlogPaths = appArgs.eventlog()
-    for (path <- eventlogPaths) {
-      apps += new ApplicationInfo(appArgs, sparkSession, fileWriter, path, index)
-      index += 1
+    val tempFile = new File("src/test/resources/workload_profiling")
+    try {
+      val fileWriter = new PrintWriter(tempFile)
```

Comment:
I would avoid using `PrintWriter` because `Methods in this class never throw I/O exceptions, although some of its constructors may. The client may inquire as to whether any errors have occurred by invoking checkError().`.

I would use `FileWriter` or `FileOutputStream` instead.